sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java

```diff
@@ -143,11 +159,66 @@ public JsonNode writeJsonNode(Object sourceEntity) {
             cosmosObjectNode.put("id", id);
         }
 
+        if (stripTransientFields) {
+            @SuppressWarnings("unchecked") final Class<T> domainType = (Class<T>) sourceEntity.getClass();
+            @SuppressWarnings("unchecked")
+            CosmosEntityInformation<T, Object> entityInfo = (CosmosEntityInformation<T, Object>) CosmosEntityInformation.getInstance(domainType);
+            List<String> transientFields =  entityInfo.getTransientFields();
+
+            if (!transientFields.isEmpty()) {
+                //strip fields with @Transient annotation from the JsonNode so they are not persisted.
+                cosmosObjectNode = stripTransientFields(cosmosObjectNode, transientFields);
+            }
+        }
+
         mapVersionFieldToEtag(sourceEntity, cosmosObjectNode);
 
         return cosmosObjectNode;
     }
 
+    /**
+     * To repopulate any transient fields that were stripped from the original object
+     * @param <T> type of source entity
+     * @param originalObjectToSave must not be {@literal null}
+     * @param responseItem must not be {@literal null}
+     * @return CosmosItemProperties
+     * @throws MappingException no mapping metadata for entity type
+     * @throws CosmosAccessException fail to map document value
+     */
+
+    public <T> JsonNode repopulateAnyTransientFields(Object originalObjectToSave, JsonNode responseItem) {
+        @SuppressWarnings("unchecked") final Class<T> domainType = (Class<T>) originalObjectToSave.getClass();
+        @SuppressWarnings("unchecked")
+        CosmosEntityInformation<T, Object> entityInfo = (CosmosEntityInformation<T, Object>) CosmosEntityInformation.getInstance(domainType);
+        List<String> transientFields =  entityInfo.getTransientFields();
+        JsonNode updatedItem = responseItem.deepCopy();
+        Boolean anyTransientFieldsSet = false;
+        if (!transientFields.isEmpty()) {
+            JsonNode originalObject = writeJsonNode(originalObjectToSave, false);
+            anyTransientFieldsSet = transientFields.stream().anyMatch(originalObject::hasNonNull);
+            if (anyTransientFieldsSet) {
+                for (Iterator<String> it = originalObject.fieldNames(); it.hasNext();) {
+                    String fieldName = it.next();
+                    ((ObjectNode) updatedItem).set(fieldName, originalObject.get(fieldName));
+                }
+            }
+        }
+        return updatedItem;
+    }
+
+    private ObjectNode stripTransientFields(ObjectNode objectNode, List<String> transientFields) {
+        Iterator<String> fieldNames = objectNode.fieldNames();
+        while (fieldNames.hasNext()) {
+            String fieldName = fieldNames.next();
+            if (transientFields.contains(fieldName)) {
+                fieldNames.remove();
```

Comment:
We don't need to do fieldNames.remove(), this is just a local variable so there is no gain from the removal.

Reply:
Yeah, will clean up this method.

Reply:
Cleaned up, much simpler :-)